components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/AuthnDataPublisherProxy.java

```diff
@@ -49,13 +60,22 @@
      */
     public void publishAuthenticationStepSuccess(HttpServletRequest request, AuthenticationContext context,
                                                  Map<String, Object> params) {
+
+        Event event = createEvent(request, context, null, params, EventName.AUTHENTICATION_STEP_SUCCESS);
+        event.addEventProperty(EventProperty.AUTHENTICATION_STATUS, AuthenticatorStatus.PASS);
+        doPublishEvent(event);
         for (AuthenticationDataPublisher publisher : dataPublishers) {
-            if (publisher.isEnabled(context) && publisher.canHandle(context)) {
+            if (isPublisherEnabledAndCanHandleContext(context, publisher)) {
                 publisher.publishAuthenticationStepSuccess(request, context, params);
             }
         }
     }
 
+    private boolean isPublisherEnabledAndCanHandleContext(AuthenticationContext context, AuthenticationDataPublisher publisher) {
```

Comment:
Shall we make this protected ? If somebody really wants he can alter is enable logic depending on a context parameter.

Reply:
Changed to protected.